In [1]:
import pandas as pd
import numpy as np

# Data analysis of dataset(SHARK TANK INDIA)

In [2]:
data = pd.read_csv('/kaggle/input/shark-tank-india-dataset/Shark Tank India Dataset.csv')

In [3]:
data

episode_number  pitch_number           brand_name  \
0                 1             1  BluePine Industries   
1                 1             2        Booz scooters   
2                 1             3  Heart up my Sleeves   
3                 2             4           Tagz Foods   
4                 2             5       Head and Heart   
..              ...           ...                  ...   
112              34           113        Green Protein   
113              34           114              On2Cook   
114              35           115        Jain Shikanji   
115              35           116                Woloo   
116              35           117         Elcare India   

                                              idea  deal  pitcher_ask_amount  \
0                                     Frozen Momos     1                50.0   
1    Renting e-bike for mobility in private spaces     1                40.0   
2                               Detachable Sleeves     1                25.0   
3                             Healthy Potato Chips     1                70.0   
4                         Brain Development Course     0                50.0   
..                                             ...   ...                 ...   
112                            Plant-Based Protein     0                60.0   
113                         Fastest Cooking Device     0               100.0   
114                                       Lemonade     1                40.0   
115                                Washroom Finder     0                50.0   
116                           Carenting for Elders     0               100.0   

     ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0           5.0        1000.00         75.0        16.00  ...             1   
1          15.0         266.67         40.0        50.00  ...             1   
2          10.0         250.00         25.0        30.00  ...             0   
3           1.0        7000.00         70.0         2.75  ...             1   
4           5.0        1000.00          0.0         0.00  ...             0   
..          ...            ...          ...          ...  ...           ...   
112         2.0        3000.00          0.0         0.00  ...             0   
113         1.0       10000.00          0.0         0.00  ...             0   
114         8.0         500.00         40.0        30.00  ...             1   
115         4.0        1250.00          0.0         0.00  ...             0   
116         2.5        4000.00          0.0         0.00  ...             0   

     anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0              0          1            0             1            0   
1              0          0            0             1            0   
2              1          0            0             1            0   
3              0          0            0             0            0   
4              0          0            0             0            0   
..           ...        ...          ...           ...          ...   
112            0          0            0             0            0   
113            0          0            0             0            0   
114            1          1            0             1            0   
115            0          0            0             0            0   
116            0          0            0             0            0   

     ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0              0                      3              25.0          5.333333  
1              0                      2              20.0         25.000000  
2              0                      2              12.5         15.000000  
3              0                      1              70.0          2.750000  
4              0                      0               0.0          0.000000  
..           ...                    ...               ...               ...  

# Number of rows and columns:

In [4]:
data.shape

(117, 28)

# Name of  the columns in the data:

In [5]:
data.columns

Index(['episode_number', 'pitch_number', 'brand_name', 'idea', 'deal',
       'pitcher_ask_amount', 'ask_equity', 'ask_valuation', 'deal_amount',
       'deal_equity', 'deal_valuation', 'ashneer_present', 'anupam_present',
       'aman_present', 'namita_present', 'vineeta_present', 'peyush_present',
       'ghazal_present', 'ashneer_deal', 'anupam_deal', 'aman_deal',
       'namita_deal', 'vineeta_deal', 'peyush_deal', 'ghazal_deal',
       'total_sharks_invested', 'amount_per_shark', 'equity_per_shark'],
      dtype='object')

# Index of the dataframe:

In [6]:
data.index

RangeIndex(start=0, stop=117, step=1)

# Columns datatype:

In [7]:

data.dtypes

episode_number             int64
pitch_number               int64
brand_name                object
idea                      object
deal                       int64
pitcher_ask_amount       float64
ask_equity               float64
ask_valuation            float64
deal_amount              float64
deal_equity              float64
deal_valuation           float64
ashneer_present            int64
anupam_present             int64
aman_present               int64
namita_present             int64
vineeta_present            int64
peyush_present             int64
ghazal_present             int64
ashneer_deal               int64
anupam_deal                int64
aman_deal                  int64
namita_deal                int64
vineeta_deal               int64
peyush_deal                int64
ghazal_deal                int64
total_sharks_invested      int64
amount_per_shark         float64
equity_per_shark         float64
dtype: object

# Basic information

In [8]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 117 entries, 0 to 116
Data columns (total 28 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   episode_number         117 non-null    int64  
 1   pitch_number           117 non-null    int64  
 2   brand_name             117 non-null    object 
 3   idea                   117 non-null    object 
 4   deal                   117 non-null    int64  
 5   pitcher_ask_amount     117 non-null    float64
 6   ask_equity             117 non-null    float64
 7   ask_valuation          117 non-null    float64
 8   deal_amount            117 non-null    float64
 9   deal_equity            117 non-null    float64
 10  deal_valuation         117 non-null    float64
 11  ashneer_present        117 non-null    int64  
 12  anupam_present         117 non-null    int64  
 13  aman_present           117 non-null    int64  
 14  namita_present         117 non-null    int64  
 15  vineet

# Missing values

In [9]:
data.isnull().sum()

episode_number           0
pitch_number             0
brand_name               0
idea                     0
deal                     0
pitcher_ask_amount       0
ask_equity               0
ask_valuation            0
deal_amount              0
deal_equity              0
deal_valuation           0
ashneer_present          0
anupam_present           0
aman_present             0
namita_present           0
vineeta_present          0
peyush_present           0
ghazal_present           0
ashneer_deal             0
anupam_deal              0
aman_deal                0
namita_deal              0
vineeta_deal             0
peyush_deal              0
ghazal_deal              0
total_sharks_invested    0
amount_per_shark         0
equity_per_shark         0
dtype: int64

# Duplicate rows

In [10]:
data[data.duplicated()]

Empty DataFrame
Columns: [episode_number, pitch_number, brand_name, idea, deal, pitcher_ask_amount, ask_equity, ask_valuation, deal_amount, deal_equity, deal_valuation, ashneer_present, anupam_present, aman_present, namita_present, vineeta_present, peyush_present, ghazal_present, ashneer_deal, anupam_deal, aman_deal, namita_deal, vineeta_deal, peyush_deal, ghazal_deal, total_sharks_invested, amount_per_shark, equity_per_shark]
Index: []

[0 rows x 28 columns]

findings:
-  There is no column with null values.
-  There are no duplicate rows.

# Description of the data:

In [11]:
#the count ,mean,std,min,25%,50%,75%,max of whole data
data.describe(include = "all").T

count unique                  top freq         mean  \
episode_number         117.0    NaN                  NaN  NaN    18.735043   
pitch_number           117.0    NaN                  NaN  NaN         59.0   
brand_name               117    117  BluePine Industries    1          NaN   
idea                     117    117         Frozen Momos    1          NaN   
deal                   117.0    NaN                  NaN  NaN     0.555556   
pitcher_ask_amount     117.0    NaN                  NaN  NaN   319.854709   
ask_equity             117.0    NaN                  NaN  NaN     5.188034   
ask_valuation          117.0    NaN                  NaN  NaN  3852.462479   
deal_amount            117.0    NaN                  NaN  NaN    31.982915   
deal_equity            117.0    NaN                  NaN  NaN     8.963504   
deal_valuation         117.0    NaN                  NaN  NaN   467.104872   
ashneer_present        117.0    NaN                  NaN  NaN     0.837607   
anupam_present         117.0    NaN                  NaN  NaN          1.0   
aman_present           117.0    NaN                  NaN  NaN     0.837607   
namita_present         117.0    NaN                  NaN  NaN     0.905983   
vineeta_present        117.0    NaN                  NaN  NaN     0.564103   
peyush_present         117.0    NaN                  NaN  NaN     0.752137   
ghazal_present         117.0    NaN                  NaN  NaN     0.222222   
ashneer_deal           117.0    NaN                  NaN  NaN     0.179487   
anupam_deal            117.0    NaN                  NaN  NaN     0.205128   
aman_deal              117.0    NaN                  NaN  NaN     0.239316   
namita_deal            117.0    NaN                  NaN  NaN     0.188034   
vineeta_deal           117.0    NaN                  NaN  NaN     0.128205   
peyush_deal            117.0    NaN                  NaN  NaN     0.230769   
ghazal_deal            117.0    NaN                  NaN  NaN     0.059829   
total_sharks_invested  117.0    NaN                  NaN  NaN     1.230769   
amount_per_shark       117.0    NaN                  NaN  NaN    18.132481   
equity_per_shark       117.0    NaN                  NaN  NaN      5.58359   

                                std      min     25%     50%      75%  \
episode_number            10.070778      1.0    10.0    19.0     27.0   
pitch_number              33.919021      1.0    30.0    59.0     88.0   
brand_name                      NaN      NaN     NaN     NaN      NaN   
idea                            NaN      NaN     NaN     NaN      NaN   
deal                       0.499041      0.0     0.0     1.0      1.0   
pitcher_ask_amount      2767.842777  0.00101    45.0    50.0     80.0   
ask_equity                 3.892121     0.25     2.5     5.0      7.5   
ask_valuation          11931.601957     0.01  666.67  1250.0  2857.14   
deal_amount               36.687391      0.0     0.0    25.0     50.0   
deal_equity               13.106769      0.0     0.0     3.0     15.0   
deal_valuation           919.988864      0.0     0.0   100.0    500.0   
ashneer_present            0.370397      0.0     1.0     1.0      1.0   
anupam_present                  0.0      1.0     1.0     1.0      1.0   
aman_present               0.370397      0.0     1.0     1.0      1.0   
namita_present             0.293108      0.0     1.0     1.0      1.0   
vineeta_present            0.498007      0.0     0.0     1.0      1.0   
peyush_present             0.433629      0.0     1.0     1.0      1.0   
ghazal_present             0.417528      0.0     0.0     0.0      0.0   
ashneer_deal                0.38541      0.0     0.0     0.0      0.0   
anupam_deal                0.405532      0.0     0.0     0.0      0.0   
aman_deal                  0.428501      0.0     0.0     0.0      0.0   
namita_deal                 0.39242      0.0     0.0     0.0      0.0   
vineeta_deal               0.335756      0.0     0.0     0.0      0.0   
peyush_d

# Categorical data

In [12]:
#the count,unique ,top and frequency of catagorical columns of data:
data.describe(include = "object").T

count unique                  top freq
brand_name   117    117  BluePine Industries    1
idea         117    117         Frozen Momos    1

# Numerical data

In [13]:
#the count ,mean,std,min,25%,50%,75% and max of numerical columns of data
data.describe().T

count         mean           std      min     25%  \
episode_number         117.0    18.735043     10.070778  1.00000   10.00   
pitch_number           117.0    59.000000     33.919021  1.00000   30.00   
deal                   117.0     0.555556      0.499041  0.00000    0.00   
pitcher_ask_amount     117.0   319.854709   2767.842777  0.00101   45.00   
ask_equity             117.0     5.188034      3.892121  0.25000    2.50   
ask_valuation          117.0  3852.462479  11931.601957  0.01000  666.67   
deal_amount            117.0    31.982915     36.687391  0.00000    0.00   
deal_equity            117.0     8.963504     13.106769  0.00000    0.00   
deal_valuation         117.0   467.104872    919.988864  0.00000    0.00   
ashneer_present        117.0     0.837607      0.370397  0.00000    1.00   
anupam_present         117.0     1.000000      0.000000  1.00000    1.00   
aman_present           117.0     0.837607      0.370397  0.00000    1.00   
namita_present         117.0     0.905983      0.293108  0.00000    1.00   
vineeta_present        117.0     0.564103      0.498007  0.00000    0.00   
peyush_present         117.0     0.752137      0.433629  0.00000    1.00   
ghazal_present         117.0     0.222222      0.417528  0.00000    0.00   
ashneer_deal           117.0     0.179487      0.385410  0.00000    0.00   
anupam_deal            117.0     0.205128      0.405532  0.00000    0.00   
aman_deal              117.0     0.239316      0.428501  0.00000    0.00   
namita_deal            117.0     0.188034      0.392420  0.00000    0.00   
vineeta_deal           117.0     0.128205      0.335756  0.00000    0.00   
peyush_deal            117.0     0.230769      0.423137  0.00000    0.00   
ghazal_deal            117.0     0.059829      0.238190  0.00000    0.00   
total_sharks_invested  117.0     1.230769      1.410457  0.00000    0.00   
amount_per_shark       117.0    18.132481     23.588682  0.00000    0.00   
equity_per_shark       117.0     5.583590     10.803799  0.00000    0.00   

                           50%      75%        max  
episode_number           19.00    27.00      35.00  
pitch_number             59.00    88.00     117.00  
deal                      1.00     1.00       1.00  
pitcher_ask_amount       50.00    80.00   30000.00  
ask_equity                5.00     7.50      25.00  
ask_valuation          1250.00  2857.14  120000.00  
deal_amount              25.00    50.00     150.00  
deal_equity               3.00    15.00      75.00  
deal_valuation          100.00   500.00    6666.67  
ashneer_present           1.00     1.00       1.00  
anupam_present            1.00     1.00       1.00  
aman_present              1.00     1.00       1.00  
namita_present            1.00     1.00       1.00  
vineeta_present           1.00     1.00       1.00  
peyush_present            1.00     1.00       1.00  
ghazal_present            0.00     0.00       1.00  
ashneer_deal              0.00     0.00       1.00  
anupam_deal               0.00     0.00       1.00  
aman_deal                 0.00     0.00       1.00  
namita_deal               0.00     0.00       1.00  
vineeta_deal              0.00     0.00       1.00  
peyush_deal               0.00     0.00       1.00  
ghazal_deal               0.00     0.00       1.00  
total_sharks_invested     1.00     2.00       5.00  
amount_per_shark         10.00    25.00     100.00  
equity_per_shark          1.25     6.00      75.00

# Max, min , mean of asked amount,asked equity, asked valuation--

In [14]:
data[["pitcher_ask_amount","ask_equity","ask_valuation"]].agg([max,min,"mean"])

pitcher_ask_amount  ask_equity  ask_valuation
max         30000.000000   25.000000  120000.000000
min             0.001010    0.250000       0.010000
mean          319.854709    5.188034    3852.462479

# Max , min,mean of deal equity,deal_amount,deal_valuation--

In [15]:
data[["deal_equity","deal_amount","deal_valuation"]].agg([max,min,"mean"])

deal_equity  deal_amount  deal_valuation
max     75.000000   150.000000     6666.670000
min      0.000000     0.000000        0.000000
mean     8.963504    31.982915      467.104872

# Max & Min asked- equity,asked-valuation and asked amount episode-wise

In [16]:
data.groupby(['episode_number'])[['pitcher_ask_amount', 'ask_equity','ask_valuation']].agg(['max', 'min'])

pitcher_ask_amount           ask_equity       ask_valuation  \
                              max       min        max   min           max   
episode_number                                                               
1                            50.0  25.00000       15.0  5.00       1000.00   
2                            70.0  50.00000        5.0  1.00       7000.00   
3                           100.0  50.00000        4.0  0.25      40000.00   
4                            75.0  50.00000       10.0  4.00       1875.00   
5                           100.0  10.00000       20.0  1.00      10000.00   
6                           100.0  45.00000       10.0  1.00      10000.00   
7                           100.0  50.00000        7.5  1.00       7500.00   
8                            56.0  30.00000        7.5  2.50       2000.00   
9                           100.0  50.00000        5.0  2.50       2000.00   
10                           30.0  25.00000        5.0  2.00       1500.00   
11                        30000.0  30.00000       25.0  5.00     120000.00   
12                           75.0  40.00000        5.0  3.00       1875.00   
13                           50.0  30.00000       10.0  2.00       2500.00   
14                          100.0  45.00000        5.0  3.00       3333.33   
15                           50.0   5.00000        5.0  3.00       1250.00   
16                           80.0  45.00000        7.0  2.00       2250.00   
17                          150.0  50.00000       10.0  3.00       5000.00   
18                          100.0  50.00000        4.0  1.00      10000.00   
19                          125.0  15.00000        8.0  1.25      10000.00   
20                           65.0  35.00000        5.0  2.00       3250.00   
21                          100.0  35.00000       10.0  5.00       1250.00   
22                           80.0  50.00000        5.0  2.00       3000.00   
23                          100.0  30.00000        5.0  1.00      10000.00   
24                           40.0  20.00000       10.0  8.00        500.00   
25                          150.0  50.00000        4.0  2.00       7500.00   
26                           65.0  50.00000       10.0  1.00       6500.00   
27                          100.0   0.00101       10.0  1.00      10000.00   
28                           90.0  50.00000        5.0  4.00       1875.00   
29                          100.0  75.00000        5.0  3.00       2857.14   
30                          300.0  50.00000       15.0  1.00      30000.00   
31                           75.0  50.00000       10.0  2.00       3750.00   
32                          200.0  35.00000        7.0  1.00       5000.00   
33                           40.0  35.00000       10.0  1.00       3500.00   
34                          100.0  30.00000       10.0  1.00      10000.00   
35                          100.0  40.00000        8.0  2.50       4000.00   

                         
                    min  
episode_number           
1                250.00  
2               1000.00  
3               1875.00  
4                500.00  
5                 50.00  
6                500.00  
7                666.67  
8                746.67  
9               1000.00  
10               500.00  
11               300.00  
12               800.00  
13               500.00  
14               900.00  
15               100.00  
16              1071.43  
17               500.00  
18              1250.00  
19               300.00  
20               700.00  
21               470.00  
22              1500.00  
23               600.00  
24               200.00  
25              1250.00  
26               500.00  
27                 0.01  
28              1000.00  
29              1500.00  
30               500.00  
31               500.00  
32               583.33  
33               400.00  
34               400.00  
35               500.00

# Max & Min deal equity,deal amount and deal valuation episode-wise

In [17]:
data.groupby(['episode_number'])[['deal_amount', 'deal_equity','deal_valuation']].agg(['max', 'min'])

deal_amount           deal_equity       deal_valuation        
                       max       min         max   min            max     min
episode_number                                                               
1                     75.0  25.00000       50.00  16.0         468.75   80.00
2                     70.0   0.00000        2.75   0.0        2545.45    0.00
3                     75.0   0.00000       15.00   0.0        1250.00    0.00
4                     75.0   0.00000       25.00   0.0        1000.00    0.00
5                    100.0   0.00000        1.50   0.0        6666.67    0.00
6                    100.0   0.00000       20.00   0.0         666.67    0.00
7                    100.0   0.00000        4.00   0.0        2500.00    0.00
8                     56.0  30.00000       33.30   2.5        2000.00  168.17
9                     50.0   0.00000       10.00   0.0         714.29    0.00
10                    30.0  25.00000       20.00   3.0        1000.00  125.00
11                    50.0   0.00000       20.00   0.0         333.33    0.00
12                    75.0   0.00000       24.00   0.0         500.00    0.00
13                   105.0  50.00000       20.00   3.0        3500.00  250.00
14                   100.0   0.00000        6.00   0.0        1666.67    0.00
15                   100.0   0.00005       40.00   5.0         250.00    0.00
16                    75.0   0.00000       15.00   0.0        1428.57    0.00
17                   150.0   0.00000       25.00   0.0        1000.00    0.00
18                     1.0   0.00000        1.00   0.0         100.00    0.00
19                    60.0   0.00000       10.00   0.0         600.00    0.00
20                   100.0   0.00000       10.00   0.0        1333.33    0.00
21                   100.0  25.00000       75.00  15.0         666.67   33.33
22                    60.0   0.00000        4.00   0.0        1500.00    0.00
23                    50.0   0.00000       10.00   0.0         500.00    0.00
24                    50.0   0.00000       40.00   0.0         142.86    0.00
25                   100.0   0.00000       15.00   0.0        1666.67    0.00
26                    65.0   0.00000       50.00   0.0        2166.67    0.00
27                   100.0   0.00000        4.00   0.0        2500.00    0.00
28                   100.0   0.00000       30.00   0.0         666.67    0.00
29                    75.0   0.00000       21.00   0.0         357.14    0.00
30                    80.0   0.00000        6.00   0.0        1333.33    0.00
31                    75.0   0.00000        2.68   0.0        2798.51    0.00
32                    50.0   0.00000        2.00   0.0        2500.00    0.00
33                    60.0   0.00000       25.00   0.0         600.00    0.00
34                    40.0   0.00000       20.00   0.0         200.00    0.00
35                    40.0   0.00000       30.00   0.0         133.33    0.00

#  Brand names where Pitcher asked amount equals to deal amount-

In [18]:
data.iloc[np.where(data["pitcher_ask_amount"]  == data["deal_amount"])][["brand_name","pitcher_ask_amount","deal_amount"]]

brand_name  pitcher_ask_amount  deal_amount
1                     Booz scooters            40.00000     40.00000
2               Heart up my Sleeves            25.00000     25.00000
3                        Tagz Foods            70.00000     70.00000
7                         Peeschute            75.00000     75.00000
9                             Cosiq            50.00000     50.00000
11                           Bummer            75.00000     75.00000
12                      Revamp Moto           100.00000    100.00000
16                     Menstrupedia            50.00000     50.00000
18               Raising Superstars           100.00000    100.00000
21                     Beyond Snack            50.00000     50.00000
22   Vivalyf Innovations- Easy Life            56.00000     56.00000
23                    Motion Breeze            30.00000     30.00000
24                            Altor            50.00000     50.00000
25                            Ariro            50.00000     50.00000
27                          Nuutjob            25.00000     25.00000
28                         Meatyour            30.00000     30.00000
29                        EventBeep            30.00000     30.00000
31         ARRCOAT Surface Textures            50.00000     50.00000
32                            Farda            30.00000     30.00000
33                   Auli Lifestyle            75.00000     75.00000
35                             LOKA            40.00000     40.00000
37                        Caragreen            50.00000     50.00000
39                The Renal Project           100.00000    100.00000
47                     Beyond Water            75.00000     75.00000
48                        Let's Try            45.00000     45.00000
49            Find Your Kicks India            50.00000     50.00000
50                    Aas Vidyalaya           150.00000    150.00000
52                       RoadBounce            80.00000     80.00000
64                       Get a Whey           100.00000    100.00000
66                  The Quirky Nari            35.00000     35.00000
67                   Hair Originals            60.00000     60.00000
78                        PawsIndia            50.00000     50.00000
79              Sunfox Technologies           100.00000    100.00000
81                  Isak Fragrances            50.00000     50.00000
83                      Rare Planet            65.00000     65.00000
85               Watt Technovations             0.00101      0.00101
87               Insurance Samadhan           100.00000    100.00000
90         Gold Safe Solutions Ind.            50.00000     50.00000
91                      Wakao Foods            75.00000     75.00000
95                     Kabaddi Adda            80.00000     80.00000
100                        AyuRythm            75.00000     75.00000
104                      GrowFitter            50.00000     50.00000
106                   Colour Me Mad            40.00000     40.00000
110              Nomad Food Project            40.00000     40.00000
114                   Jain Shikanji            40.00000     40.00000

# Brand names where Ask_equity equals to deal_equity

In [19]:
data.iloc[np.where(data["ask_equity"] == data["deal_equity"])][["brand_name","ask_equity","deal_equity"]]

brand_name  ask_equity  deal_equity
21  Beyond Snack         2.5          2.5
44       Cocofit         5.0          5.0
55          Otua         1.0          1.0

# Brand names where Ask valuation equals to deal valuation

In [20]:
data.iloc[np.where(data["ask_valuation"]  == data["deal_valuation"])][["brand_name","ask_valuation","deal_valuation"]]

brand_name  ask_valuation  deal_valuation
21  Beyond Snack         2000.0          2000.0

# Episodes  in which all female sharks are present--

In [21]:
data[(data["vineeta_present"] == 1) & (data["namita_present"] == 1) & (data["ghazal_present"] == 1)][["episode_number","vineeta_present","namita_present","ghazal_present"]]

episode_number  vineeta_present  namita_present  ghazal_present
75               24                1               1               1
76               24                1               1               1
77               24                1               1               1
78               25                1               1               1
79               25                1               1               1
80               25                1               1               1
81               26                1               1               1
82               26                1               1               1
83               26                1               1               1
84               27                1               1               1
85               27                1               1               1
86               27                1               1               1
87               27                1               1               1
88               28                1               1               1
89               28                1               1               1
90               28                1               1               1
91               29                1               1               1
92               29                1               1               1
93               29                1               1               1
110              34                1               1               1
111              34                1               1               1
112              34                1               1               1
113              34                1               1               1
114              35                1               1               1
115              35                1               1               1
116              35                1               1               1

# Episodes  in which all male sharks are present--

In [22]:
data[(data["ashneer_present"] == 1) & (data["aman_present"] == 1) & (data["anupam_present"] == 1) & (data["peyush_present"] == 1)][["ashneer_present","aman_present","anupam_present","peyush_present"]]

ashneer_present  aman_present  anupam_present  peyush_present
21                 1             1               1               1
22                 1             1               1               1
23                 1             1               1               1
24                 1             1               1               1
25                 1             1               1               1
..               ...           ...             ...             ...
112                1             1               1               1
113                1             1               1               1
114                1             1               1               1
115                1             1               1               1
116                1             1               1               1

[69 rows x 4 columns]

# Episodes and brand names where Mr.Ashneer present--

In [23]:
data.iloc[np.where(data['ashneer_present'] == 1)][["episode_number",'brand_name', 'ashneer_present']]

episode_number           brand_name  ashneer_present
0                 1  BluePine Industries                1
1                 1        Booz scooters                1
2                 1  Heart up my Sleeves                1
3                 2           Tagz Foods                1
4                 2       Head and Heart                1
..              ...                  ...              ...
112              34        Green Protein                1
113              34              On2Cook                1
114              35        Jain Shikanji                1
115              35                Woloo                1
116              35         Elcare India                1

[98 rows x 3 columns]

# Mr.Ashneer present count--

In [24]:
data[['ashneer_present']] [data['ashneer_present'] == 1].count()

ashneer_present    98
dtype: int64

# Episodes and brand names where Mr.Anupam present--

In [25]:
data.iloc[np.where(data['anupam_present'] == 1)][["episode_number",'brand_name', 'anupam_present']]

episode_number           brand_name  anupam_present
0                 1  BluePine Industries               1
1                 1        Booz scooters               1
2                 1  Heart up my Sleeves               1
3                 2           Tagz Foods               1
4                 2       Head and Heart               1
..              ...                  ...             ...
112              34        Green Protein               1
113              34              On2Cook               1
114              35        Jain Shikanji               1
115              35                Woloo               1
116              35         Elcare India               1

[117 rows x 3 columns]

# Mr.Anupam present count--

In [26]:
data[['anupam_present']] [data['anupam_present'] == 1].count()

anupam_present    117
dtype: int64

# Episodes and brand names where Mr.Aman is  present--

In [27]:
data.iloc[np.where(data['aman_present'] == 1)][["episode_number",'brand_name', 'aman_present']]

episode_number           brand_name  aman_present
0                 1  BluePine Industries             1
1                 1        Booz scooters             1
2                 1  Heart up my Sleeves             1
3                 2           Tagz Foods             1
4                 2       Head and Heart             1
..              ...                  ...           ...
112              34        Green Protein             1
113              34              On2Cook             1
114              35        Jain Shikanji             1
115              35                Woloo             1
116              35         Elcare India             1

[98 rows x 3 columns]

# Mr.Aman present count--

In [28]:
data[['aman_present']] [data['aman_present'] == 1].count()

aman_present    98
dtype: int64

# Episodes and brand names where Mr.Peyush is  present--

In [29]:
data.iloc[np.where(data['peyush_present'] == 1)][["episode_number",'brand_name', 'peyush_present']]

episode_number                      brand_name  peyush_present
21                8                    Beyond Snack               1
22                8  Vivalyf Innovations- Easy Life               1
23                8                   Motion Breeze               1
24                9                           Altor               1
25                9                           Ariro               1
..              ...                             ...             ...
112              34                   Green Protein               1
113              34                         On2Cook               1
114              35                   Jain Shikanji               1
115              35                           Woloo               1
116              35                    Elcare India               1

[88 rows x 3 columns]

# Mr.Peyush present count--

In [30]:
data[['peyush_present']] [data['peyush_present'] == 1].count()

peyush_present    88
dtype: int64

# Episodes and brand names where Ms.Namita is  present--

In [31]:
data.iloc[np.where(data['namita_present'] == 1)][["episode_number",'brand_name', 'namita_present']]

episode_number           brand_name  namita_present
0                 1  BluePine Industries               1
1                 1        Booz scooters               1
2                 1  Heart up my Sleeves               1
3                 2           Tagz Foods               1
4                 2       Head and Heart               1
..              ...                  ...             ...
112              34        Green Protein               1
113              34              On2Cook               1
114              35        Jain Shikanji               1
115              35                Woloo               1
116              35         Elcare India               1

[106 rows x 3 columns]

# Ms.Namita present count--

In [32]:
data[['namita_present']] [data['namita_present'] == 1].count()

namita_present    106
dtype: int64

# Episodes and brand names where Ms.Vineeta is  present--

In [33]:
data.iloc[np.where(data['vineeta_present'] == 1)][["episode_number",'brand_name', 'vineeta_present']]

episode_number           brand_name  vineeta_present
0                 1  BluePine Industries                1
1                 1        Booz scooters                1
2                 1  Heart up my Sleeves                1
3                 2           Tagz Foods                1
4                 2       Head and Heart                1
..              ...                  ...              ...
112              34        Green Protein                1
113              34              On2Cook                1
114              35        Jain Shikanji                1
115              35                Woloo                1
116              35         Elcare India                1

[66 rows x 3 columns]

# Ms.Vineeta present count--

In [34]:
data[['vineeta_present']] [data['vineeta_present'] == 1].count()

vineeta_present    66
dtype: int64

# Episodes and brand names where Ms.ghazal is  present--

In [35]:
data.iloc[np.where(data['ghazal_present'] == 1)][["episode_number",'brand_name', 'ghazal_present']]

episode_number                brand_name  ghazal_present
75               24              The Sass Bar               1
76               24               KG Agrotech               1
77               24            Nuskha Kitchen               1
78               25                 PawsIndia               1
79               25       Sunfox Technologies               1
80               25                    Alpino               1
81               26           Isak Fragrances               1
82               26          Julaa Automation               1
83               26               Rare Planet               1
84               27              Theka Coffee               1
85               27        Watt Technovations               1
86               27       Aliste Technologies               1
87               27        Insurance Samadhan               1
88               28                  Humpy A2               1
89               28              Kunafa World               1
90               28  Gold Safe Solutions Ind.               1
91               29               Wakao Foods               1
92               29                PDD Falcon               1
93               29                PlayBox TV               1
110              34        Nomad Food Project               1
111              34               Twee in One               1
112              34             Green Protein               1
113              34                   On2Cook               1
114              35             Jain Shikanji               1
115              35                     Woloo               1
116              35              Elcare India               1

# Ms.Ghazal present count--

In [36]:
data[['ghazal_present']] [data['ghazal_present'] == 1].count()

ghazal_present    26
dtype: int64

findings:
-  Mr.Anupam is present in all episodes.
-  Mr.Peyush is present in  88  episodes
-  Mr.Ashneer is present in    episodes
-  Mr.Aman is present in  98  episodes
-  Ms.Namita is present in 106   episodes
-  Ms.vineeta is present in  66  episodes
-  Ms.Ghazal is present in  26 episodes

# Brand names where sharks become co-founder--

In [37]:
# dataframe showing the sharks who become co_founder by getting equity greater than or equals to 50%:
co_founder = data.iloc[np.where(data["deal_equity"] >= 50)][["brand_name","deal_equity","total_sharks_invested"]]

In [38]:
co_founder

brand_name  deal_equity  total_sharks_invested
1     Booz scooters         50.0                      2
65    Sid07 Designs         75.0                      1
81  Isak Fragrances         50.0                      1

# Mr.Ashneer deals--

In [39]:
data.iloc[np.where((data['deal'] == 1) & (data['ashneer_deal'] == 1))][['brand_name','ashneer_deal',"deal_amount"]]

brand_name  ashneer_deal  deal_amount
0      BluePine Industries             1         75.0
1            Booz scooters             1         40.0
3               Tagz Foods             1         70.0
15             Skippi Pops             1        100.0
21            Beyond Snack             1         50.0
23           Motion Breeze             1         30.0
29               EventBeep             1         30.0
38         The Yarn Bazaar             1        100.0
45            Bamboo India             1         50.0
49   Find Your Kicks India             1         50.0
50           Aas Vidyalaya             1        150.0
55                    Otua             1          1.0
58                 WeSTOCK             1         60.0
63                IN A CAN             1        100.0
64              Get a Whey             1        100.0
67          Hair Originals             1         60.0
108             Tweek Labs             1         60.0
109                 Proxgy             1         10.0
110     Nomad Food Project             1         40.0
114          Jain Shikanji             1         40.0

# Total of Mr.Ashneer deals--

In [40]:
data[['ashneer_deal']] [data['ashneer_deal'] == 1].count()

ashneer_deal    21
dtype: int64

# Mr.Aman deals--

In [41]:
data.iloc[np.where((data['deal'] == 1) & (data['aman_deal'] == 1))][['brand_name','aman_deal',"deal_amount"]]

brand_name  aman_deal  deal_amount
0      BluePine Industries          1     75.00000
7                Peeschute          1     75.00000
11                  Bummer          1     75.00000
12             Revamp Moto          1    100.00000
15             Skippi Pops          1    100.00000
21            Beyond Snack          1     50.00000
24                   Altor          1     50.00000
25                   Ariro          1     50.00000
27                 Nuutjob          1     25.00000
28                Meatyour          1     30.00000
29               EventBeep          1     30.00000
32                   Farda          1     30.00000
35                    LOKA          1     40.00000
38         The Yarn Bazaar          1    100.00000
39       The Renal Project          1    100.00000
42        Hammer Lifestyle          1    100.00000
44                 Cocofit          1      0.00005
47            Beyond Water          1     75.00000
48               Let's Try          1     45.00000
49   Find Your Kicks India          1     50.00000
58                 WeSTOCK          1     60.00000
63                IN A CAN          1    100.00000
64              Get a Whey          1    100.00000
71            Namhya Foods          1     50.00000
100               AyuRythm          1     75.00000
104             GrowFitter          1     50.00000
114          Jain Shikanji          1     40.00000

# Total deals of Mr.Aman--

In [42]:
data[['aman_deal']] [data['aman_deal'] == 1].count()

aman_deal    28
dtype: int64

# Mr.Anupam deals--

In [43]:
data.iloc[np.where((data['deal'] == 1) & (data['anupam_deal'] == 1))][['brand_name','anupam_deal',"deal_amount"]]

brand_name  anupam_deal  deal_amount
2               Heart up my Sleeves            1     25.00000
9                             Cosiq            1     50.00000
12                      Revamp Moto            1    100.00000
15                      Skippi Pops            1    100.00000
22   Vivalyf Innovations- Easy Life            1     56.00000
28                         Meatyour            1     30.00000
31         ARRCOAT Surface Textures            1     50.00000
35                             LOKA            1     40.00000
36                            Annie            1    105.00000
37                        Caragreen            1     50.00000
38                  The Yarn Bazaar            1    100.00000
44                          Cocofit            1      0.00005
45                     Bamboo India            1     50.00000
48                        Let's Try            1     45.00000
49            Find Your Kicks India            1     50.00000
63                         IN A CAN            1    100.00000
66                  The Quirky Nari            1     35.00000
67                   Hair Originals            1     60.00000
75                     The Sass Bar            1     50.00000
78                        PawsIndia            1     50.00000
79              Sunfox Technologies            1    100.00000
85               Watt Technovations            1      0.00101
108                      Tweek Labs            1     60.00000
114                   Jain Shikanji            1     40.00000

# Total deals of Mr.Anupam--

In [44]:
data[['anupam_deal']] [data['anupam_deal'] == 1].count()

anupam_deal    24
dtype: int64

# Mr.Peyush deals--

In [45]:
data.iloc[np.where((data['deal'] == 1) & (data['peyush_deal'] == 1))][['brand_name','peyush_deal',"deal_amount"]]

brand_name  peyush_deal  deal_amount
22   Vivalyf Innovations- Easy Life            1     56.00000
25                            Ariro            1     50.00000
27                          Nuutjob            1     25.00000
28                         Meatyour            1     30.00000
29                        EventBeep            1     30.00000
35                             LOKA            1     40.00000
36                            Annie            1    105.00000
37                        Caragreen            1     50.00000
38                  The Yarn Bazaar            1    100.00000
43                              PNT            1     25.00000
49            Find Your Kicks India            1     50.00000
50                    Aas Vidyalaya            1    150.00000
52                       RoadBounce            1     80.00000
58                          WeSTOCK            1     60.00000
61                  The State Plate            1     40.00000
63                         IN A CAN            1    100.00000
65                    Sid07 Designs            1     25.00000
67                   Hair Originals            1     60.00000
76                      KG Agrotech            1     10.00000
79              Sunfox Technologies            1    100.00000
81                  Isak Fragrances            1     50.00000
85               Watt Technovations            1      0.00101
87               Insurance Samadhan            1    100.00000
88                         Humpy A2            1    100.00000
90         Gold Safe Solutions Ind.            1     50.00000
108                      Tweek Labs            1     60.00000
109                          Proxgy            1     10.00000

# Total deals of Mr.Peyush--

In [46]:
data[['peyush_deal']] [data['peyush_deal'] == 1].count()

peyush_deal    27
dtype: int64

# Ms.Namita deals--

In [47]:
data.iloc[np.where((data['deal'] == 1) & (data['namita_deal'] == 1))][['brand_name','namita_deal',"deal_amount"]]

brand_name  namita_deal  deal_amount
11                  Bummer            1     75.00000
15             Skippi Pops            1    100.00000
16            Menstrupedia            1     50.00000
24                   Altor            1     50.00000
27                 Nuutjob            1     25.00000
32                   Farda            1     30.00000
33          Auli Lifestyle            1     75.00000
36                   Annie            1    105.00000
39       The Renal Project            1    100.00000
44                 Cocofit            1      0.00005
47            Beyond Water            1     75.00000
49   Find Your Kicks India            1     50.00000
50           Aas Vidyalaya            1    150.00000
58                 WeSTOCK            1     60.00000
63                IN A CAN            1    100.00000
79     Sunfox Technologies            1    100.00000
83             Rare Planet            1     65.00000
85      Watt Technovations            1      0.00101
91             Wakao Foods            1     75.00000
95            Kabaddi Adda            1     80.00000
106          Colour Me Mad            1     40.00000
110     Nomad Food Project            1     40.00000

# Total deals of Ms.Namita--

In [48]:
data[['namita_deal']] [data['namita_deal'] == 1].count()

namita_deal    22
dtype: int64

# Ms.Vineeta deals--

In [49]:
data.iloc[np.where((data['deal'] == 1) & (data['vineeta_deal'] == 1))][['brand_name','vineeta_deal',"deal_amount"]]

brand_name  vineeta_deal  deal_amount
0         BluePine Industries             1         75.0
1               Booz scooters             1         40.0
2         Heart up my Sleeves             1         25.0
8                        NOCD             1         20.0
9                       Cosiq             1         50.0
15                Skippi Pops             1        100.0
64                 Get a Whey             1        100.0
66            The Quirky Nari             1         35.0
79        Sunfox Technologies             1        100.0
88                   Humpy A2             1        100.0
90   Gold Safe Solutions Ind.             1         50.0
91                Wakao Foods             1         75.0
95               Kabaddi Adda             1         80.0
110        Nomad Food Project             1         40.0
114             Jain Shikanji             1         40.0

# Total deals of Ms.Vineeta--

In [50]:
data[['vineeta_deal']] [data['vineeta_deal'] == 1].count()

vineeta_deal    15
dtype: int64

# Ms.Ghazal deals--

In [51]:
data.iloc[np.where((data['deal'] == 1) & (data['ghazal_deal'] == 1))][['brand_name','ghazal_deal',"deal_amount"]]

brand_name  ghazal_deal  deal_amount
75               The Sass Bar            1     50.00000
79        Sunfox Technologies            1    100.00000
85         Watt Technovations            1      0.00101
88                   Humpy A2            1    100.00000
90   Gold Safe Solutions Ind.            1     50.00000
91                Wakao Foods            1     75.00000
110        Nomad Food Project            1     40.00000

# Total deals of Ms.Ghazal--

In [52]:
data[['ghazal_deal']] [data['ghazal_deal'] == 1].count()

ghazal_deal    7
dtype: int64

findings:
-  out of 117 ashneer done 21   deals
-  out of 117 vineeta done 15  deals
-  out of 117 namita done 22    deals
-  out of 117 aman done  28 deals
-  out of 117 ghazal done  7 deals
-  out of 117 peyush done  27 deals
-  out of 117 anupam done  24 deals


# Brand names who got a deal--

In [53]:
data[['brand_name']] [data['deal'] == 1]

brand_name
0    BluePine Industries
1          Booz scooters
2    Heart up my Sleeves
3             Tagz Foods
7              Peeschute
..                   ...
106        Colour Me Mad
108           Tweek Labs
109               Proxgy
110   Nomad Food Project
114        Jain Shikanji

[65 rows x 1 columns]

# Total number of brand names who got a deal--

In [54]:
data[['brand_name']] [data['deal'] == 1].agg(['count'])

brand_name
count          65

# Total  amount invested in the show--

In [55]:
data[["deal_amount"]].agg(["sum"])

deal_amount
sum   3742.00106

# Total  amount invested by Mr.Aman--

In [56]:
data[['amount_per_shark']] [data['aman_deal'] == 1].agg(['sum'])

amount_per_shark
sum        887.500017

# Total  amount invested by Mr.Anupam--

In [57]:
data[['amount_per_shark']] [data['anupam_deal'] == 1].agg(['sum'])

amount_per_shark
sum        533.833603

# Total  amount invested by Mr.Peyush--

In [58]:
data[['amount_per_shark']] [data['peyush_deal'] == 1].agg(['sum'])

amount_per_shark
sum        719.666919

# Total  amount invested by Mr.Ashneer--

In [59]:
data[['amount_per_shark']] [data['ashneer_deal'] == 1].agg(['sum'])

amount_per_shark
sum        494.333333

# Total  amount invested by Ms.Vineeta--

In [60]:
data[['amount_per_shark']] [data['vineeta_deal'] == 1].agg(['sum'])

amount_per_shark
sum        328.333333

# Total  amount invested by Ms.Ghazal--

In [61]:
data[['amount_per_shark']] [data['ghazal_deal'] == 1].agg(['sum'])

amount_per_shark
sum        130.000252

# Total  amount invested by Ms.Namita--

In [62]:
data[['amount_per_shark']] [data['namita_deal'] == 1].agg(['sum'])

amount_per_shark
sum        648.333603

# Maximun & Minimum number of deals episode-wise

In [63]:
data.groupby('episode_number')['deal'].value_counts().agg(['max', 'min'])

max    3
min    1
Name: deal, dtype: int64

# Maximum deals amount on brands--

In [64]:
data[data['deal_amount'] == data['deal_amount'].max()][['brand_name', 'idea', 'deal_amount']]

brand_name        idea  deal_amount
50  Aas Vidyalaya  EdTech App        150.0

# Minimum deals amount on brand--

In [65]:
data.iloc[np.where(((data['deal'] == 1) & (data['deal_amount'] == data['deal_amount'].min())))][['brand_name', 'idea', 'deal_amount']]

brand_name                                  idea  deal_amount
19   Torch-it  Gadgets for visually impaired people          0.0

# Brand names in which 5 or more sharks invested--

In [66]:
#dataframe showing deals in which 5 or more sharks are invested:
data.iloc[np.where(data["total_sharks_invested"] >= 5)][["brand_name","total_sharks_invested"]]

brand_name  total_sharks_invested
15            Skippi Pops                      5
49  Find Your Kicks India                      5
63               IN A CAN                      5
79    Sunfox Technologies                      5

findings:
- There is no brand where more than 5 sharks are invested.

# Brand names in which only one shark invested 

In [67]:
#dataframe showing deals in which only one shark invested:
data.iloc[np.where(data["total_sharks_invested"] == 1)][["brand_name","total_sharks_invested"]]

brand_name  total_sharks_invested
3                  Tagz Foods                      1
7                   Peeschute                      1
8                        NOCD                      1
16               Menstrupedia                      1
23              Motion Breeze                      1
31   ARRCOAT Surface Textures                      1
33             Auli Lifestyle                      1
42           Hammer Lifestyle                      1
43                        PNT                      1
52                 RoadBounce                      1
55                       Otua                      1
61            The State Plate                      1
65              Sid07 Designs                      1
71               Namhya Foods                      1
76                KG Agrotech                      1
78                  PawsIndia                      1
81            Isak Fragrances                      1
83                Rare Planet                      1
87         Insurance Samadhan                      1
100                  AyuRythm                      1
104                GrowFitter                      1
106             Colour Me Mad                      1

# Brand names in which 2 ,3 or 4 sharks are invested-

In [68]:
#dataframe showing those brands name who get sharks between 2 to 4:
data[(data["total_sharks_invested"] >=2) & (data["total_sharks_invested"] <=5)][["brand_name","total_sharks_invested"]]

brand_name  total_sharks_invested
0               BluePine Industries                      3
1                     Booz scooters                      2
2               Heart up my Sleeves                      2
9                             Cosiq                      2
11                           Bummer                      2
12                      Revamp Moto                      2
15                      Skippi Pops                      5
18               Raising Superstars                      2
21                     Beyond Snack                      2
22   Vivalyf Innovations- Easy Life                      2
24                            Altor                      2
25                            Ariro                      2
27                          Nuutjob                      3
28                         Meatyour                      3
29                        EventBeep                      3
32                            Farda                      2
35                             LOKA                      3
36                            Annie                      3
37                        Caragreen                      2
38                  The Yarn Bazaar                      4
39                The Renal Project                      2
44                          Cocofit                      3
45                     Bamboo India                      2
47                     Beyond Water                      2
48                        Let's Try                      2
49            Find Your Kicks India                      5
50                    Aas Vidyalaya                      3
58                          WeSTOCK                      4
63                         IN A CAN                      5
64                       Get a Whey                      3
66                  The Quirky Nari                      2
67                   Hair Originals                      3
75                     The Sass Bar                      2
79              Sunfox Technologies                      5
85               Watt Technovations                      4
88                         Humpy A2                      3
90         Gold Safe Solutions Ind.                      3
91                      Wakao Foods                      3
95                     Kabaddi Adda                      2
108                      Tweek Labs                      3
109                          Proxgy                      2
110              Nomad Food Project                      4
114                   Jain Shikanji                      4

# Episode wise minimum and maximum of sharks invested-

In [69]:
data.groupby(["episode_number"])["total_sharks_invested"].agg([max,min])

max  min
episode_number          
1                 3    2
2                 1    0
3                 1    0
4                 2    0
5                 2    0
6                 5    0
7                 2    0
8                 2    1
9                 2    0
10                3    3
11                2    0
12                3    0
13                4    2
14                2    0
15                3    1
16                2    0
17                5    0
18                1    0
19                4    0
20                5    0
21                3    1
22                3    0
23                1    0
24                2    0
25                5    0
26                1    0
27                4    0
28                3    0
29                3    0
30                2    0
31                1    0
32                1    0
33                3    0
34                4    0
35                4    0

# Minimum and Maximum sharks invested in each brand:

In [70]:
data.groupby(["brand_name"])["total_sharks_invested"].agg([max,min])

max  min
brand_name                              
ARRCOAT Surface Textures          1    1
Aas Vidyalaya                     3    3
Agro tourism                      0    0
Aliste Technologies               0    0
Alpino                            0    0
...                             ...  ...
Vivalyf Innovations- Easy Life    2    2
Wakao Foods                       3    3
Watt Technovations                4    4
WeSTOCK                           4    4
Woloo                             0    0

[117 rows x 2 columns]

# Rename of column(idea)-

In [71]:
data=  data.rename(columns = {"idea":"purpose"})

# Brand names who ask for 1 crore and got a deal--

In [72]:
# dataframe showing those brand names in which pitcher_ask_amt and deal amount == 100:
data[(data["pitcher_ask_amount"] == 100) & (data["deal_amount"] == 100)][["brand_name","pitcher_ask_amount","deal_amount"]]

brand_name  pitcher_ask_amount  deal_amount
12          Revamp Moto               100.0        100.0
18   Raising Superstars               100.0        100.0
39    The Renal Project               100.0        100.0
64           Get a Whey               100.0        100.0
79  Sunfox Technologies               100.0        100.0
87   Insurance Samadhan               100.0        100.0

# Brand names who asked equity between 0 to 50--

In [73]:
# dataframe showing those brand names in which ask_equity is between 0 to 50:
data[(data["ask_equity"]  >= 0) & (data["ask_equity"] <=50)][["brand_name","ask_equity"]]

brand_name  ask_equity
0    BluePine Industries         5.0
1          Booz scooters        15.0
2    Heart up my Sleeves        10.0
3             Tagz Foods         1.0
4         Head and Heart         5.0
..                   ...         ...
112        Green Protein         2.0
113              On2Cook         1.0
114        Jain Shikanji         8.0
115                Woloo         4.0
116         Elcare India         2.5

[117 rows x 2 columns]

# Brand names where deal equity is between 0 to 50--

In [74]:
# dataframe showing those brand names in which deal_equity is between 0 to 50:
data[(data["deal_equity"]  >= 0) & (data["deal_equity"] <=50)]

episode_number  pitch_number           brand_name  \
0                 1             1  BluePine Industries   
1                 1             2        Booz scooters   
2                 1             3  Heart up my Sleeves   
3                 2             4           Tagz Foods   
4                 2             5       Head and Heart   
..              ...           ...                  ...   
112              34           113        Green Protein   
113              34           114              On2Cook   
114              35           115        Jain Shikanji   
115              35           116                Woloo   
116              35           117         Elcare India   

                                           purpose  deal  pitcher_ask_amount  \
0                                     Frozen Momos     1                50.0   
1    Renting e-bike for mobility in private spaces     1                40.0   
2                               Detachable Sleeves     1                25.0   
3                             Healthy Potato Chips     1                70.0   
4                         Brain Development Course     0                50.0   
..                                             ...   ...                 ...   
112                            Plant-Based Protein     0                60.0   
113                         Fastest Cooking Device     0               100.0   
114                                       Lemonade     1                40.0   
115                                Washroom Finder     0                50.0   
116                           Carenting for Elders     0               100.0   

     ask_equity  ask_valuation  deal_amount  deal_equity  ...  ashneer_deal  \
0           5.0        1000.00         75.0        16.00  ...             1   
1          15.0         266.67         40.0        50.00  ...             1   
2          10.0         250.00         25.0        30.00  ...             0   
3           1.0        7000.00         70.0         2.75  ...             1   
4           5.0        1000.00          0.0         0.00  ...             0   
..          ...            ...          ...          ...  ...           ...   
112         2.0        3000.00          0.0         0.00  ...             0   
113         1.0       10000.00          0.0         0.00  ...             0   
114         8.0         500.00         40.0        30.00  ...             1   
115         4.0        1250.00          0.0         0.00  ...             0   
116         2.5        4000.00          0.0         0.00  ...             0   

     anupam_deal  aman_deal  namita_deal  vineeta_deal  peyush_deal  \
0              0          1            0             1            0   
1              0          0            0             1            0   
2              1          0            0             1            0   
3              0          0            0             0            0   
4              0          0            0             0            0   
..           ...        ...          ...           ...          ...   
112            0          0            0             0            0   
113            0          0            0             0            0   
114            1          1            0             1            0   
115            0          0            0             0            0   
116            0          0            0             0            0   

     ghazal_deal  total_sharks_invested  amount_per_shark  equity_per_shark  
0              0                      3              25.0          5.333333  
1              0                      2              20.0         25.000000  
2              0                      2              12.5         15.000000  
3              0                      1              70.0          2.750000  
4              0                      0               0.0          0.000000  
..           ...                    ...               ...               ...  

# Brand name who asked amount of 3cr--

In [75]:
#dataframe showing deals in which they ask valuation of 1200cr:
data.iloc[np.where(data["pitcher_ask_amount"] == 30000)][["brand_name","pitcher_ask_amount"]]

brand_name  pitcher_ask_amount
30  Gopal's 56             30000.0

# New column of length of brand name--

In [76]:
#adding a new column of length of the brand name:
data["length_of_brand_name"] = data["brand_name"].apply(lambda x : len(x))

In [77]:
data.shape

(117, 29)

# New column of comparision between ask_equity and deal_equity--

In [78]:
data['difference_equity'] = data.apply(lambda x: x['deal_equity'] - x['ask_equity'], axis = 1)

In [79]:
data[["ask_equity","deal_equity","difference_equity"]]

ask_equity  deal_equity  difference_equity
0           5.0        16.00              11.00
1          15.0        50.00              35.00
2          10.0        30.00              20.00
3           1.0         2.75               1.75
4           5.0         0.00              -5.00
..          ...          ...                ...
112         2.0         0.00              -2.00
113         1.0         0.00              -1.00
114         8.0        30.00              22.00
115         4.0         0.00              -4.00
116         2.5         0.00              -2.50

[117 rows x 3 columns]

# New column of comparision between ask_valuation and deal_valuation--

In [80]:
#diference_valuation (deal_valuation- (asked_valuation))
data['difference_valuation'] = data.apply(lambda x: x['deal_valuation'] - x['ask_valuation'], axis = 1)
data[["ask_valuation","deal_valuation","difference_valuation"]]

ask_valuation  deal_valuation  difference_valuation
0          1000.00          468.75               -531.25
1           266.67           80.00               -186.67
2           250.00           83.33               -166.67
3          7000.00         2545.45              -4454.55
4          1000.00            0.00              -1000.00
..             ...             ...                   ...
112        3000.00            0.00              -3000.00
113       10000.00            0.00             -10000.00
114         500.00          133.33               -366.67
115        1250.00            0.00              -1250.00
116        4000.00            0.00              -4000.00

[117 rows x 3 columns]

# New column of comparision between ask_amount and deal_amount--

In [81]:
data['difference_amount'] = data.apply(lambda x: x['deal_amount'] - x['pitcher_ask_amount'], axis = 1)
data[["pitcher_ask_amount","deal_amount","difference_amount"]]

pitcher_ask_amount  deal_amount  difference_amount
0                  50.0         75.0               25.0
1                  40.0         40.0                0.0
2                  25.0         25.0                0.0
3                  70.0         70.0                0.0
4                  50.0          0.0              -50.0
..                  ...          ...                ...
112                60.0          0.0              -60.0
113               100.0          0.0             -100.0
114                40.0         40.0                0.0
115                50.0          0.0              -50.0
116               100.0          0.0             -100.0

[117 rows x 3 columns]